tests/PHPUnit/Integration/DataAccess/ArchiveSelectorTest.php

```diff
@@ -30,24 +30,85 @@ protected static function configureFixture($fixture)
     }
 
     /**
-     * @dataProvider getTestDataForGetArchiveIdAndVisits
+     * @dataProvider getTestDataForGetArchiveIds
      */
-    public function test_getArchiveIdAndVisits_returnsCorrectResult($archiveRows, $segment, $minDateProcessed, $includeInvalidated, $expected)
+    public function test_getArchiveIds_returnsCorrectResult($archiveRows, $siteIds, $periods, $segment, $plugins, $expected)
     {
         Fixture::createWebsite('2010-02-02 00:00:00');
+        Fixture::createWebsite('2010-02-02 00:00:00');
+
+        foreach ($periods as $index => [$periodStr, $dateStr]) {
+            $periods[$index] = Factory::build($periodStr, $dateStr);
+        }
 
         $this->insertArchiveData($archiveRows);
 
-        $params = new Parameters(new Site(1), Factory::build('day', '2019-10-05'), new Segment($segment, [1]));
-        $result = ArchiveSelector::getArchiveIdAndVisits($params, $minDateProcessed, $includeInvalidated);
+        $archiveIds = ArchiveSelector::getArchiveIds($siteIds, $periods, new Segment($segment, $siteIds), $plugins);
 
-        if ($result[4] !== false) {
-            Date::factory($result[4]);
-        }
+        $this->assertEquals($expected, $archiveIds);
+    }
+
+    public function getTestDataForGetArchiveIds()
+    {
+        return [
+            // normal single site
+            [
+                [
+                    ['idarchive' => 1, 'idsite' => 1, 'period' => 1, 'date1' => '2020-03-01', 'date2' => '2020-03-01', 'name' => 'done', 'value' => 1],
+                    ['idarchive' => 2, 'idsite' => 1, 'period' => 2, 'date1' => '2020-03-02', 'date2' => '2020-03-08', 'name' => 'done', 'value' => 4],
 
-        unset($result[4]);
+                    ['idarchive' => 3, 'idsite' => 1, 'period' => 1, 'date1' => '2020-03-01', 'date2' => '2020-03-01', 'name' => 'done', 'value' => 2],
+                    ['idarchive' => 4, 'idsite' => 1, 'period' => 2, 'date1' => '2020-02-24', 'date2' => '2020-03-01', 'name' => 'done', 'value' => 2],
 
-        $this->assertEquals($expected, $result);
+                    ['idarchive' => 5, 'idsite' => 1, 'period' => 1, 'date1' => '2020-03-03', 'date2' => '2020-03-03', 'name' => 'done', 'value' => 1],
+                    ['idarchive' => 6, 'idsite' => 2, 'period' => 2, 'date1' => '2020-02-24', 'date2' => '2020-03-01', 'name' => 'done', 'value' => 1],
+                ],
+                [1],
+                [
+                    ['day', '2020-03-01'],
+                    ['week', '2020-03-02'],
+                ],
+                '',
+                [],
+                [
+                    'done' => [
+                        '2020-03-01,2020-03-01' => [1],
```

Comment:
Maybe I didn't understand the code correctly, but shouldn't that contain the biggest archive id available for that period. Looking at the data above, that seems to be `3` and not `1`

Reply:
The value for `3` is `DONE_ERROR` so we don't use it.

Reply:
ah ok. that makes sense